ros2cli/ros2cli/node/daemon.py

```diff
@@ -39,18 +39,18 @@ def __init__(self, args):
     @property
     def connected(self):
         try:
-            self._proxy.system.listMethods()
+            self._methods = [
+                method
+                for method in self._proxy.system.listMethods()
+                if not method.startswith('system.')
+            ]
```

Comment:
save the method list at the same time here, when it confirms the connection XMLRPC server. this guarantees method list is updated during connectivity check.